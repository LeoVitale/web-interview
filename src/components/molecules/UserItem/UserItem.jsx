import React from 'react'
import PropTypes from 'prop-types'
import Avatar from 'components/atoms/Avatar'
import UserHeader from 'components/atoms/UserHeader'
import Icon from 'components/atoms/Icon'
import styles from './UserItem.module.scss'

const UserItem = ({ action, user }) => {
  const { userItem, userData, actions, addIcon, addAvatar } = styles
  let userHeader
  const name = user ? `${user.firstName} ${user.lastName}` : ''
  if (user) {
    userHeader = <UserHeader title={name} subheader="Teste" />
  } else {
    userHeader = <UserHeader title="Add Family Member" />
  }
  return (
    <div className={userItem}>
      <div className={userData}>
        {user ? (
          <Avatar name={name} img={user.avatar} />
        ) : (
          <Avatar
            className={addAvatar}
            icon={<Icon type="add" className={addIcon} />}
          />
        )}
        {userHeader}
      </div>
      <div className={actions}>{action}</div>
    </div>
  )
}

UserItem.propTypes = {}

export default UserItem
