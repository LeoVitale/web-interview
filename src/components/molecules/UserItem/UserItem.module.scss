.userItem {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 20px;
  .userData {
    display: flex;
    align-items: center;
  }
}
