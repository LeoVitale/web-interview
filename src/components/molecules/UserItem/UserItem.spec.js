import React from 'react';
import { mount } from 'enzyme';
import UserItem from './UserItem.jsx';

describe('<UserItem />', () => {
  const userItem = mount(<UserItem />);

  it('should render UserItem as expected', () => {
    expect(userItem).toMatchSnapshot();
  });
});
