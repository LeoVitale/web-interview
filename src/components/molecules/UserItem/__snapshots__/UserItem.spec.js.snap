// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserItem /> should render UserItem as expected 1`] = `
<UserItem>
  <div
    className="userItem"
  >
    <div
      className="userData"
    >
      <Avatar>
        <div
          className="avatar"
        >
          <img
            alt=""
            src="https://react.semantic-ui.com/images/avatar/small/elliot.jpg"
          />
        </div>
      </Avatar>
      <UserHeader
        subheader="Teste"
        title="Leonardo Vitale"
      >
        <div
          className="userHeader"
        >
          <span
            className="title"
          >
            Leonardo Vitale
          </span>
          <span
            className="subheader"
          >
            Teste
          </span>
        </div>
      </UserHeader>
    </div>
    <div
      className="actions"
    >
      change
    </div>
  </div>
</UserItem>
`;
